tldraw/tldraw
# feat: add support for embedding near.social widgets

This pull request adds support for embedding near.social widgets.

https://github.com/tldraw/tldraw/assets/16282460/8ea8351e-7f30-4da7-968f-1376e06dff74

Near Social Widgets are ~React components stored on-chain that are interpreted by the [VM](https://github.com/NearSocial/VM) on render. These components are sandboxed, composable, customizable, and already integrated with social features like commenting, liking, starring (bookmarking)... with the ability to build much more. And they typically cost less than 10 cents to deploy forever.

Widgets can be created, updated, and see those changes reflect on the canvas:

https://github.com/tldraw/tldraw/assets/16282460/e3573f3a-bcdf-48b4-8a44-fd02ad9883a1

### Change Type


[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Test Plan

1. Checkout this branch
2. Install packages
3. Embed a near.social component, any widget you find [here will work](https://near.social/#/mob.near/widget/LastWidgets)


(I did not see tests for existing embeds)


### Release Notes

- Adds support for embedding near.social widgets.
